### PR TITLE
Add options for read and write operations.

### PR DESCRIPTION
**Pull request by qwandor**
I've left out `prepare-authorize` for now because I don't understand how it's used.

### CODE REVIEW

File: bluez-async/src/lib.rs [hunk 1]
```diff
@@ -699,3 +812,11 @@ impl BluetoothSession {
             .flat_map(|message| stream::iter(BluetoothEvent::message_to_events(message))))
     }
 }
+
+fn options_for_offset(offset: usize) -> PropMap {
```
Comment (alsuren) [inline bluez-async/src/lib.rs:816]: `options_propmap_for_offset()`
`options_propmap_from_offset()`
`offset_to_propmap()`

File: bluez-async/src/lib.rs [hunk 2]
```diff
@@ -611,43 +673,94 @@ impl BluetoothSession {
...
+    pub async fn write_descriptor_value_with_offset(
+        &self,
+        id: &DescriptorId,
+        value: impl Into<Vec<u8>>,
+        offset: usize,
     ) -> Result<(), BluetoothError> {
         let descriptor = self.descriptor(id);
-        Ok(descriptor.write_value(value.into(), HashMap::new()).await?)
+        Ok(descriptor
+            .write_value(value.into(), options_for_offset(offset))
```
Comment (alsuren) [inline bluez-async/src/lib.rs:762]: I guess .write_value() is generated code?

Comment (qwandor) [reply bluez-async/src/lib.rs:762]: Yes.

File: bluez-async/src/lib.rs (hunk 1 above)
Comment (qwandor) [reply bluez-async/src/lib.rs]: Renamed.

**Review by alsuren — APPROVED** [review]

**Pull request merged**